Remove generated_unboxing_wrappers and setManuallyBoxedKernel

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49613 Remove backward and requires_grad from Autograd backend key
* #49259 Remove `use_c10_dispatcher: full` lines
* **#49251 Remove generated_unboxing_wrappers and setManuallyBoxedKernel**
* #49220 Remove .impl_UNBOXED() and functionalities associated with it
* #49164 Remove codegen logic to support non-c10-full ops

Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

smessmer committed:
Remove generated_unboxing_wrappers and setManuallyBoxedKernel

Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]

bhosmer commented on aten/src/ATen/core/boxing/KernelFunction_impl.h:
```diff
@@ -168,14 +168,4 @@ inline std::enable_if_t<!guts::is_stateless_lambda<std::decay_t<Lambda>>::value,
-    }
-    TORCH_INTERNAL_ASSERT(boxed_kernel_func_ == nullptr, "Tried to set a manually boxed kernel for a kernel that already has a boxed kernel set.");
-    TORCH_INTERNAL_ASSERT(unboxed_kernel_func_ != nullptr, "Tried to set a manually boxed kernel for an invalid KernelFunction.");
-    boxed_kernel_func_ = func;
```
IIRC this is the only place `boxed_kernel_func_` was ever updated after construction, can it now be a const member?

smessmer replied:
maybe if we remove the assignment operator from KernelFunction. But currently, that operator implicitly exists and is used in some call sites.

bhosmer reviewed (approved):
![image](https://user-images.githubusercontent.com/355313/103743192-b4bc2680-4fb0-11eb-8932-cd5b485077ca.png)
😍

smessmer committed:
Update on "Remove generated_unboxing_wrappers and setManuallyBoxedKernel"


Since all ops are c10-full and use templated unboxing now, we don't need to codegenerate any unboxing logic anymore.
Since this codegen was the only code using setManuallyBoxedKernel, we can also remove that functionality from KernelFunction, OperatorEntry and Dispatcher.

Differential Revision: [D25502865](https://our.internmc.facebook.com/intern/diff/D25502865/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25502865/)!

[ghstack-poisoned]
